webapp/views/App/views/Data/MapView/CoordinateAttributeDataLayer/CoordinateAttributePolygon.js

```diff
@@ -0,0 +1,245 @@
+import React, { useState } from 'react'
+import { Rectangle, Circle } from 'react-leaflet'
+import GeometryUtil from 'leaflet-geometryutil'
+import L from 'leaflet'
+import { useMapContextOptions } from '@webapp/components/Map/MapContext'
+import { useSurveyInfo } from '@webapp/store/survey'
+import _ from 'lodash'
+import PropTypes from 'prop-types'
+
+export const CoordinateAttributePolygon = (props) => {
+  const surveyInfo = useSurveyInfo()
+  const polygonProperties = JSON.parse(surveyInfo.props.samplingPolygon)
+
+  const options = useMapContextOptions()
+  const { showSamplingPolygon, showControlPoints, showPlotReferencePoint } = options
+  const SamplingPolygonOpoacity = showSamplingPolygon ? 1 : 0
+
+  const { latitude, longitude } = props
+
+  const len_lat_meters = Number(polygonProperties.length_latitude) || 0
+  const len_lng_meters = Number(polygonProperties.length_longitude) || 0
+  const offset_north_meters = Number(polygonProperties.offset_north) || 0
+  const offset_east_meters = Number(polygonProperties.offset_east) || 0
+  const numberOfPointsNorth = Number(polygonProperties.number_of_points_north) || 0
+  const numberOfPointsEast = Number(polygonProperties.number_of_points_east) || 0
+  const numberOfPointsCircle = Number(polygonProperties.number_of_points_circle) || 0
+  const controlPointOffsetNorh = Number(polygonProperties.controlpoint_offset_north) || 0
```

Comment:
controlpointOffsetNorth or control_point_offset_north?

Reply:
a simple question: since the polygon properties are internal to Arena, can the name of the polygon properties be in camel case? and is it possible to use a module with getters and updaters to access and modify those properties?